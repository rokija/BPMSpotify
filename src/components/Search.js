import React, { Component } from 'react';
import Searchbar from '../containers/SearchBar';
import TrackList from '../containers/TrackList';
import BeforeTrackList from './BeforeTrackList';

export default class Search extends Component {
    constructor(){
        super();

        this.setSearchQuery = this.setSearchQuery.bind(this);
        this.state = {
            searchQuery: ''
        };
    }

    setSearchQuery (searchQueryValue) {
        this.setState({
            searchQuery: searchQueryValue
        });
    }

    render() {
        return (
            <div className="search-page">
                <Searchbar setSearchQuery={this.setSearchQuery} />
                { this.state.searchQuery === '' ? <BeforeTrackList /> : <TrackList searchQuery={this.state.searchQuery}/> }
            </div>
        );
    }
}
