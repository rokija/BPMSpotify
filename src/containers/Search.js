import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from "prop-types";
import { bindActionCreators } from "redux";
import { getSearchResults } from '../actions/searchActions';
import '../styles/about-page.css';

class Search extends Component {
    constructor() {
        super();

        this.state = {
            term: ''
        };

        this.onInputChange = this.onInputChange.bind(this);
        this.onFormSubmit = this.onFormSubmit.bind(this);
    }

    onInputChange(event) {
        this.setState({ term: event.target.value });
    }

    onFormSubmit(event) {
        event.preventDefault();

        this.props.getSearchResults(this.state.term);
        this.setState({ term: '' });
    }

    render() {
        return (
            <div>
                <form>
                    <input
                        placeholder="Search a song"
                        value={this.state.term}
                        onChange={this.onInputChange} />
                    <button onClick={this.onFormSubmit} type="submit" >Search</button>
                </form>
            </div>
        );
    }
}


Search.propTypes = {
    dispatch: PropTypes.func,
    searchTerm: PropTypes.string,
    getSearchResults: PropTypes.func,
};

Search.contextTypes = {
    router: PropTypes.object
};


function mapDispatchToProps(dispatch) {
    return bindActionCreators({ getSearchResults }, dispatch);
}

export default connect(null, mapDispatchToProps)(Search);
