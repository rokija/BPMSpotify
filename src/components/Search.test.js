import React from 'react';
import { shallow } from 'enzyme';
import Search from './Search';

describe('Search', () => {
    const wrapper = shallow(<Search />);

    it('renders properly', () => {
        expect(wrapper.length).toEqual(1);
    });

    it('sets Search Query', () => {
        const wrapperInstance = wrapper.instance();
        wrapperInstance.setSearchQuery('some value');
        expect(wrapperInstance.state.searchQuery).toEqual('some value');
    });
});