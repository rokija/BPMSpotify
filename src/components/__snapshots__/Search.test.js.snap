// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search matches snapshot 1`] = `
<div
  className="search-page"
>
  <Connect(SearchBar)
    setSearchQuery={[Function]}
  />
  <BeforeTrackList />
</div>
`;
